linkerd/protocol/thriftmux/src/main/scala/com/twitter/finagle/buoyant/linkerd/ThriftMuxServerPrep.scala

```diff
@@ -0,0 +1,44 @@
+package com.twitter.finagle.buoyant.linkerd
+
+import com.twitter.finagle.client.StackClient
+import com.twitter.finagle.mux.{Request, Response}
+import com.twitter.finagle._
+import com.twitter.io.Buf
+import com.twitter.util.Future
+
+object ThriftMuxServerPrep {
+  val role: Stack.Role = StackClient.Role.prepConn
+
+  val module = new Stack.ModuleParams[ServiceFactory[Request, Response]] {
+    override def parameters: Seq[Stack.Param[_]] = Nil
+
+    override val role: Stack.Role = ThriftServerPrep.role
+    override val description = "Prepare ThriftMux connection"
+
+    def make(
+      params: Stack.Params,
+      next: ServiceFactory[Request, Response]
+    ): ServiceFactory[Request, Response] = {
+      val arrayBytesServiceFactory = ArrayBytesToMuxFilter.andThen(next)
+      MuxToArrayBytesFilter.andThen(ThriftServerPrep.module.make(params, arrayBytesServiceFactory))
+    }
+  }
+
+  object MuxToArrayBytesFilter extends Filter[mux.Request, mux.Response, Array[Byte], Array[Byte]] {
+    def apply(request: mux.Request, service: Service[Array[Byte], Array[Byte]]): Future[mux.Response] = {
+      val reqBytes = Buf.ByteArray.Owned.extract(request.body)
+      service(reqBytes) map { repBytes =>
```

Comment:
Tiny style nit: we usually prefer `service(reqBytes).map {` over `service(reqBytes) map {`